src/aggregate/expr/expression.c

```diff
@@ -479,6 +479,9 @@ static int rpevalNext_filter(ResultProcessor *rp, SearchResult *r) {
       return RS_RESULT_OK;
     }
 
+    // Reduce the total number of results
+    RS_ASSERT(rp->parent->totalResults > 0);
+    rp->parent->totalResults--;
```

Comment:
We should have a test where the filter is after a grouping step, to make sure the count is still correct in this case